Update build number of a rerender

blowekamp opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

blowekamp committed:
Update SimpleITK 2.1.1.1

Update for rebuild against ITK library.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

blowekamp commented:
conda-forge-admin, please rerender

blowekamp commented:
@conda-forge-admin, please rerender

blowekamp committed:
The strip option does not apply to this configuration

blowekamp committed:
Explicit use ninja as build tool in testing

blowekamp committed:
MNT: Re-rendered with conda-build 3.22.0, conda-smithy 3.21.2, and conda-forge-pinning 2022.10.03.09.31.48

blowekamp commented:
@jaimergp Do these updates look good for the simpleitk package update?

jaimergp commented on recipe/meta.yaml:
```diff
@@ -40,15 +40,16 @@ requirements:
+    - cmake --build . --config Release                                            # [not win]
 
 about:
   home: http::/www.simpleitk.org
```
```suggestion
  home: http://www.simpleitk.org
```

While you are at it, small typo :)

jaimergp commented on recipe/build.sh:
```diff
@@ -19,7 +19,6 @@ cmake \
     -D "CMAKE_INSTALL_PREFIX=${PREFIX}" \
     -D "CMAKE_PROGRAM_PATH=${BUILD_PREFIX}" \
```
The default value on conda-forge is `-DCMAKE_PROGRAM_PATH="$BUILD_PREFIX/bin;$PREFIX/bin" \
`, which is part of the `CMAKE_ARGS` env var set during compiler activation. Ideally, we should deviate as little as possible from the default configurations. A regular CMake incantation looks like:

```bash
cmake ${CMAKE_ARGS} ${PROJECT_FLAGS} ${SRC_DIR}
```

where `PROJECT_FLAGS` contains the project-specific definitions like `-D SimpleITK_BUILD_DISTRIBUTE:BOOL=ON` and friends.

I wonder if we can make it look like what we have now in https://github.com/conda-forge/simpleitk-feedstock/pull/27. If that's not possible, I like having comments next to the lines that "don't follow the rule" to make it easier for contributors in the future. That's my personal preference, at least.

blowekamp replied:
Looking at the linux build I see the following:
```
 +CMAKE_ARGS=-DCMAKE_AR=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ar -DCMAKE_CXX_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gcc-ar -DCMAKE_C_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gcc-ar -DCMAKE_RANLIB=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ranlib -DCMAKE_CXX_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gcc-ranlib -DCMAKE_C_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-gcc-ranlib -DCMAKE_LINKER=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-ld -DCMAKE_STRIP=$BUILD_PREFIX/bin/x86_64-conda-linux-gnu-strip -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_FIND_ROOT_PATH=$PREFIX;$BUILD_PREFIX/x86_64-conda-linux-gnu/sysroot -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_PROGRAM_PATH=$BUILD_PREFIX/bin;$PREFIX/bin
```

When this build script was originally developed most of these options were not set. The conda build has significantly improved in CMake project based support. Many of the options are vary similar to what was chosen for these projects.

There is clearly a need for an updates here.

blowekamp replied:
On OSX I see: 
```
CMAKE_ARGS=-DCMAKE_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_CXX_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_C_COMPILER_AR=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ar -DCMAKE_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_CXX_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_C_COMPILER_RANLIB=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ranlib -DCMAKE_LINKER=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-ld -DCMAKE_STRIP=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-strip -DCMAKE_INSTALL_NAME_TOOL=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-install_name_tool -DCMAKE_LIBTOOL=$BUILD_PREFIX/bin/x86_64-apple-darwin13.4.0-libtool -DCMAKE_OSX_DEPLOYMENT_TARGET=10.9 -DCMAKE_BUILD_TYPE=Release -DCMAKE_OSX_SYSROOT=/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk -DCMAKE_INSTALL_PREFIX=$PREFIX -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_PROGRAM_PATH=$BUILD_PREFIX/bin;$PREFIX/bin
```
Which does not include the `CMAKE_FIND_ROOT_*` options that are currently coded here or that are on linux.

blowekamp replied:
Looking over the cmake args here, it looks like OS X deployment target and sys root would be missing from the configuration for libsimpleitk, but it would be in libitk and simpleitk.

jaimergp reviewed (commented):
LGTM. Just some comments with respect to build.sh standardization.

blowekamp committed:
fix typo in about url

blowekamp committed:
Use CMAKE_ARGS is build.sh

Remove CMAKE variable values now provided in CMAKE_ARGS, keep
CMAKE_FIND_* as conda does not set these for OSX
currently.
